protocol/x/clob/abci.go

```diff
@@ -198,7 +200,27 @@ func PrepareCheckState(
 
 	// 6. Get all potentially liquidatable subaccount IDs and attempt to liquidate them.
 	subaccountIds := liquidatableSubaccountIds.GetSubaccountIds()
-	if err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, subaccountIds); err != nil {
+	numDeleveragingAttempts, err := keeper.LiquidateSubaccountsAgainstOrderbook(ctx, subaccountIds)
+	if err != nil {
+		panic(err)
+	}
+
+	// 7. Deleverage subaccounts with open positions in final settlement markets.
+	subaccountPositionInfo := map[uint32]map[bool]map[satypes.SubaccountId]struct{}{
```

Comment:
Should we have a pre-processing step that verifies the `subaccountPositionInfo` map is accurate (it could contain entries for subaccounts that have closed positions due to daemon lag right)?

Reply:
hmm, this would only be useful if the preprocessing was pretty fast right? The preprocessing would be able to save us from attempting to final settle deleverage subaccounts with no open position. I don't think it is very costly for the deleveraging operation to simply fail due to no position existing, so personally don't think we need to preprocess